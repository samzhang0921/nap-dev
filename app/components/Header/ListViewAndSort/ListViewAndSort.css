.ListViewAndSort {
  order: 1;
  align-self: baseline;
  display: flex;
}
@media (max-width: 949px) {
  .ListViewAndSort {
    order: 2;    
    align-self: baseline;
  }
}
