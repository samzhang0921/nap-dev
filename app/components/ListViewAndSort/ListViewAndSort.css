.ListViewAndSort {
  flex: 1;
  align-self: baseline;
  display: flex;
}
