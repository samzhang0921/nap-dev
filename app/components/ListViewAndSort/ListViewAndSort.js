import React, { Component } from 'react';
import style from './ListViewAndSort.css';
const ListViewAndSort = props => {
  return (
    <div className={style.ListViewAndSort}>
      ListView
    </div>
  );
};

export default ListViewAndSort;
