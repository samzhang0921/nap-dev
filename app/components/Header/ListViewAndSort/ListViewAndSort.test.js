import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import ListViewAndSort from './ListViewAndSort';

describe('ListViewAndSort', () => {

  const changeProductsView = jest.fn();
  const updateSort = jest.fn();

  it('should match its snapshot with items', () => {

    const tree = renderer.create(<ListViewAndSort changeProductsView={changeProductsView} updateSort={updateSort}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should change class on the onClick function', () => {

    const component = mount(<ListViewAndSort changeProductsView={changeProductsView} updateSort={updateSort}/>);
    component.find('select').simulate('change', {
      target: {
        value: 'Change function'
      }
    });
    expect(toJson(component)).toMatchSnapshot();
  });
});
