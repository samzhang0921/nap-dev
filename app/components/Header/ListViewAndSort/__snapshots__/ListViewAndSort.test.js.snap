// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListViewAndSort should change class on the onClick function 1`] = `
<ListViewAndSort
  changeProductsView={[Function]}
  updateSort={[Function]}
>
  <div
    className="ListViewAndSort"
  >
    <ListView
      changeProductsView={[Function]}
    >
      <div>
        <button
          className="ListView"
          onClick={[Function]}
        >
          Product View
        </button>
      </div>
    </ListView>
    <ListSort
      updateSort={[Function]}
    >
      <div
        className="sortBy"
      >
        <select
          name="sortOrder"
          onChange={[Function]}
        >
          <option
            value="sort"
          >
            Sort by
          </option>
          <option
            value="low"
          >
            Price Low to High
          </option>
          <option
            value="high"
          >
            Price High to Low
          </option>
        </select>
      </div>
    </ListSort>
  </div>
</ListViewAndSort>
`;

exports[`ListViewAndSort should match its snapshot with items 1`] = `
<div
  className="ListViewAndSort"
>
  <div>
    <button
      className="ListView"
      onClick={[Function]}
    >
      Product View
    </button>
  </div>
  <div
    className="sortBy"
  >
    <select
      name="sortOrder"
      onChange={[Function]}
    >
      <option
        value="sort"
      >
        Sort by
      </option>
      <option
        value="low"
      >
        Price Low to High
      </option>
      <option
        value="high"
      >
        Price High to Low
      </option>
    </select>
  </div>
</div>
`;
